### PR TITLE
Check for robots.txt

### PR DESCRIPTION
**Pull request by GatewayBit**
This adds another stage to check if the site has a `robots.txt` file. It doesn't check what info it contains, only if the file exists. The condition is tested based on HTTP response code received  from the server and it will follow `301` redirects. If a `robots.txt` file does exist, it will get a PASS, if not, it will get a MEH. It's absence from the server is not a security risk and will not receive a FAIL or FATAL.

### CODE REVIEW

**Comment by woodruffw** [comment]
Thanks! I'll review this tomorrow.

File: twa [hunk 1]
```diff
@@ -390,8 +390,22 @@ function stage_4_scm_repos {
...
 }
 
+# Stage 5: Check if the server provides a robots.txt file.
+#
+# Check for "The Robots Exclusion Protocol".
+# Will follow redirects to ensure file exists.
+function stage_5_robots_check {
+  verbose "Stage 5: Robots Exclusion Protocol check"
+
+  http_response_code=$(curl -L -s -o /dev/null -w "%{http_code}" "${domain}/robots.txt")
```
Comment (woodruffw) [inline twa]: You can use the `fetch_respcode` function here:

```bash
code=$(fetch_respcode "${domain}/robots.txt")
```

File: twa [hunk 2]
```diff
@@ -390,8 +390,22 @@ function stage_4_scm_repos {
...
+  if [[ ${http_response_code} == 200 ]]; then
+    PASS "Robots.txt is provided at: ${domain}/robots.txt"
+  else
```
Comment (woodruffw) [inline twa]: Let's be explicit here, and check for a 404 for the `MEH`. Then, the base `else` case can be an `UNK`.

File: twa [hunk 3]
```diff
@@ -390,8 +390,22 @@ function stage_4_scm_repos {
...
+# Stage 5: Check if the server provides a robots.txt file.
+#
+# Check for "The Robots Exclusion Protocol".
+# Will follow redirects to ensure file exists.
+function stage_5_robots_check {
+  verbose "Stage 5: Robots Exclusion Protocol check"
+
+  http_response_code=$(curl -L -s -o /dev/null -w "%{http_code}" "${domain}/robots.txt")
+  if [[ ${http_response_code} == 200 ]]; then
+    PASS "Robots.txt is provided at: ${domain}/robots.txt"
```
Comment (woodruffw) [inline twa]: Style nit: "Robots.txt" -> "robots.txt"

**Comment by woodruffw** [comment]
Three small nits, but otherwise looks good to me!

File: twa (hunk 2 above)
Comment (woodruffw) [reply twa]: (See `stage_4_scm_repos` for an example of this).

**Comment by woodruffw** [comment]
JFYI, I've moved this repo to my employer's organization, since they've offered to steward it.

**Comment by GatewayBit** [comment]
Thanks for your constructive feedback! I've applied the changes suggested from your review. Please let me know if there are any issues.

**Comment by woodruffw** [comment]
Thanks! I'll merge this in a bit.

**Comment by woodruffw** [comment]
Thanks again!

**Pull request merged**